web-platform/website/src/components/pages/account-settings/EditAccountEmailView.tsx

```diff
@@ -0,0 +1,278 @@
+import { FontSizeRaw, Fonts } from "../../../styles/Fonts";
+import { getFirebaseErrorMessage } from "../../../utils/FirebaseErrors";
+import {
+  AccountSettingsState,
+  useAccountSettingsDataContext,
+} from "./AccountSettingsDataContext";
+
+
+const strings = Strings.components.pages.accountSettings.editEmailAccount;
+const firebaseStings = Strings.utils.firebaseError.errorMessages
+
+export function EditAccountEmailView() {
+  const [confirmationView, setConfirmationView] = useState(false)
+  const [email, setEmail] = useState("");
+  const [isDisabled, setDisabled] = useState(true);
+  const [password, setPassword] = useState("");
+  const [passwordVisible, setPasswordVisbile] = useState(false);
+  const [eyeIconVisible, setVisibility] = useState(false)
+  const [validationResult, setValidationResult] =
+    useState<LoginFormValidationResult>();
+  const { setAccountSettingsState } = useAccountSettingsDataContext();
+
+
+  const isEmailInvalid = Boolean(
+    validationResult && validationResult.email && validationResult.email.error
+  );
+
+  const isPasswordInvalid = Boolean(
+    validationResult &&
+    validationResult.password &&
+    validationResult.password.error
+  );
+
+  // Prevents closure in validateAndUpdate
+  const emailRef = useRef(email);
+  emailRef.current = email;
+
+  const passwordRef = useRef(password);
+  passwordRef.current = password;
+
+  useEffect(() => {
+    setValidationResult((validationResult) => ({
+      ...(validationResult || {
+        email: { error: undefined },
+        password: { error: undefined },
+      }),
+      valid: true,
+      passwordRules: [],
+    }));
+  }, []);
+
+  const { changeUserEmail } = useAuthentication();
+
+  const handleChange = (e: any) => {
+    setValidationResult({
+      email: {},
+      password: {},
+      passwordRules: [],
+      valid: true,
+    });
+
+    setDisabled(false)
+    if (e.target.value === "") {
+      setDisabled(true)
+    }
+    if (e.target.name === "password") {
+      setPassword(e.target.value)
+      setVisibility(true)
+      return
+    }
+    if (e.target.name === "email") {
+      setEmail(e.target.value)
+      return
+    }
+  };
+
+
+  async function validateAndUpdate() {
+    setValidationResult(undefined);
+
+    const validationResult = validateLoginForm(
+      emailRef.current,
+      passwordRef.current
+    );
+
+    setValidationResult(validationResult);
+
+    try {
+      await changeUserEmail(emailRef.current, passwordRef.current);
+      setConfirmationView(true)
+    } catch (e) {
+
+      const error = getFirebaseErrorMessage(e as FirebaseError)
+      let emailErr = ""
+      let passwordErr = ""
+
+      error.indexOf(firebaseStings["auth/email-already-in-use"]) > -1 || error.indexOf(firebaseStings["auth/invalid-email"]) > -1 ? emailErr = error : emailErr = ""
+
+      error.indexOf(firebaseStings["auth/wrong-password"]) > -1 ? passwordErr = error : passwordErr = ""
+
+      setValidationResult({
```

Comment:
We can set this to be undefined here.

Reply:
This will remain because I need the unique errors for email and password fields.